graph_social_network/social.py

```diff
@@ -61,12 +83,31 @@ def getAllSocialPaths(self, userID):
         """
         visited = {}  # Note that this is a dictionary, not a set
         # !!!! IMPLEMENT ME
+        q = Queue()
+
+        q.enqueue([userID])
+
+        while q.size() > 0:
+            path = q.dequeue()
+            newUserID = path[-1]
+            if newUserID not in visited:
+                visited[newUserID] = path
+                for friendID in self.friendships[newUserID]:
+                    if friendID not in visited:
+                        new_path = list(path)
+                        new_path.append(friendID)
+                        q.enqueue(new_path)
         return visited
 
 
 if __name__ == '__main__':
     sg = SocialGraph()
-    sg.populateGraph(10, 2)
+    sg.populateGraph(1000, 5)
```

Comment:
This is good for testing question 1 in the README and visualizing different things, but might be good to change this back to (10,2) afterwards. It makes it easier to read and interpret what's happening with your code in the console.